src/pages/workspaces/workspace/WorkspaceContainer.js

```diff
@@ -37,45 +40,130 @@ const navIconProps = {
 }
 
 
-const WorkspaceTabs = pure(({ namespace, name, activeTab, refresh }) => {
-  const navTab = ({ tabName, href }) => {
-    const selected = tabName === activeTab
-    return h(Fragment, [
-      a({
-        style: { ...styles.tab, ...(selected ? styles.activeTab : {}) },
-        onClick: selected ? refresh : undefined,
-        href
-      }, tabName),
-      navSeparator
+class WorkspaceTabs extends PureComponent {
+  menu = createRef()
+
+  render() {
+    const { namespace, name, workspace, activeTab, refresh, onDelete } = this.props
+    const navTab = ({ tabName, href }) => {
+      const selected = tabName === activeTab
+      return h(Fragment, [
+        a({
+          style: { ...styles.tab, ...(selected ? styles.activeTab : {}) },
+          onClick: selected ? refresh : undefined,
+          href
+        }, tabName),
+        navSeparator
+      ])
+    }
+    return contextBar({ style: styles.tabContainer }, [
+      navSeparator,
+      navTab({ tabName: 'dashboard', href: Nav.getLink('workspace', { namespace, name }) }),
+      navTab({ tabName: 'notebooks', href: Nav.getLink('workspace-notebooks', { namespace, name }) }),
+      navTab({ tabName: 'data', href: Nav.getLink('workspace-data', { namespace, name }) }),
+      navTab({ tabName: 'jobs' }),
+      navTab({ tabName: 'history' }),
+      navTab({ tabName: 'tools', href: Nav.getLink('workspace-tools', { namespace, name }) }),
+      div({ style: { flexGrow: 1 } }),
+      h(Interactive, { as: 'div', ...navIconProps }, [icon('copy', { size: 22 })]),
+      h(PopupTrigger, {
+        ref: this.menu,
+        content: contextMenu([
+          { children: 'Share' },
+          { children: 'Publish' },
+          {
+            children: 'Delete',
+            disabled: !workspace || !_.includes(workspace.accessLevel, ['OWNER', 'PROJECT_OWNER']),
```

Comment:
Would it be possible to also include a message on click? Something like, "You must be an owner of this project or the Google project containing it."

Reply:
It's pretty clearly disabled, so people probably won't try to click it, but we could add a tooltip to describe why.

Reply:
(I'll do that in a followon)